### ISSUE TITLE
getParentIds() ignores height parameter

### ISSUE THREAD

**Issue by BobRay** [modxbot/migrate#1235]
bobray created Redmine issue ID 1235

The method returns the full list of parents up the tree, regardless of the second argument's value.
The method also won't get the parents of a document in another context (possibly by design). I'm not sure recursion is a good idea here unless it's much faster since it's a straight shot up the parent tree. This much simpler code works (and also will get parents in another context &#8211; not sure it that's good or bad), but needs some sanity checking: function getParentIds($id= null, $height= 10) {
$parentId= 0;
$parents= array (); while ($height-- && $id) { $resource = $this->getObject('modResource', array('id'=>$id));
$id = $resource->get('parent'); if (is_numeric($id)) { $parents[] = $id; } else { break; }
} return $parents;
}

**Comment by splittingred** [comment]
splittingred submitted:

That solution wont work for a few reasons - you're executing a query for each depth, which could put a huge burden on the server, since makeUrl and other functions use getParentIds. It wont get documents from another context, that is correct. I've committed a fix in r5704; it seems the problem was that PHP handles $var-- not in the C-style context, and therefore was not decrementing the $height variable. This has been fixed.
